cpp/serve/engine.cc

```diff
@@ -132,18 +157,13 @@ class Engine {
    * generation results for those finished requests.
    */
   void Step() {
-    // - Action 0. Abort requests.
-    action_abort_request_->Step(estate_);
-    // - Action 1. Prefill the front-most waiting request.
-    bool prefill_processed = action_new_request_prefill_->Step(estate_);
-    if (prefill_processed) {
-      return;
-    }
-    // - Action 2. Run decode step.
-    bool decode_processed = action_batch_decode_->Step(estate_);
-    if (decode_processed) {
-      ProcessFinishedRequest(estate_, models_, tokenizer_, max_single_sequence_length_);
-      return;
+    for (EngineAction action : actions_) {
+      Array<Request> processed_requests = action->Step(estate_);
+      if (!processed_requests.empty()) {
+        ActionStepPostProcess(processed_requests, estate_, models_, tokenizer_,
+                              max_single_sequence_length_);
+        return;
+      }
```

Comment:
@junrushao Just want to get some feedback on the change here! Does this change make sense to you?

My intention is that we initialize the list of actions when initializing the engine, and at each step, we go through the actions, find one to take, and then apply the post-process.